client/src/index.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ReadMoreModal from './components/ReadMoreModal.jsx';
+import './main.css';
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            productName: 'Nike Blazer', //default for now
+            styleName: 'Gray/Blue', //default for now
+            productDescription: '',
+            textBlock: '',
+            productPrice: 100, //default for now
+            imageUrl: 'https://static.nike.com/a/images/t_default/eric5lwitzffpoisq0rj/blazer-mid-77-vintage-shoe-flCCX4.jpg' //default for now
+        }
+        
+    } 
+
+    componentDidMount() {
+
+        //fetching my service
+        fetch(`http://localhost:5000/product/${this.props.productId}/description`)
+        .then(response => {
+            return response.json();
+        })
+        .then(data => {
+            this.setState({
+                productDescription: data.description,
+                textBlock: data.textBlock
+            })
+
+            // once I get Damien's API
+            // return fetch(`http://localhost:4000/${this.props.productId}/${this.props.style}`)
+        })
+        // .then(response => {
+        //     return response.json();
+        // })
+        // .then(data => {
+        //     this.setState({
+        //         productName: data.productName,
+        //         styleName: data.styleName,
+        //         productPrice: data.productPrice
+        //     })
+        //      
+        //     once I get Abraham's API
+        //     return fetch(`http://localhost:3000/${this.props.productId}/${this.props.style})
+        // })
+        // .then(response => {
+        //     return response.json();
+        // })
+        // .then(data => {
+        //     this.setState({
+        //         imageUrl: data.imageUrl
+        //     })
+        // })
+        .catch(error => {
+            console.error(err);
+        })
+
+    }
+
+    toggleModal() {
+        var modal = document.querySelector('.modal');
+        modal.classList.toggle("toggle-modal");
+    }
+
```

Comment:
instead of manipulating the DOM, I like to use conditional rendering, that way state/view management is attached to react, not DOM directly. Check it out here https://reactjs.org/docs/conditional-rendering.html#gatsby-focus-wrapper

Reply:
ohhh i see what you mean. like having the property of whether or not the modal view is showing, inside of the state. good idea! i totally did not think of using conditional rendering